arangosh/Shell/V8ClientConnection.cpp

```diff
@@ -158,16 +158,20 @@ void V8ClientConnection::setInterrupted(bool interrupted) {
   }
 }
 
-bool V8ClientConnection::isConnected() {
-  if (_connection) {
-    return _connection->state() == fuerte::Connection::State::Connected;
+bool V8ClientConnection::isConnected() const {
+  std::weak_ptr<fuerte::Connection> connection = _connection;
```

Comment:
wait why don't you just get the `_lock` ?
Also whyt about `requestDataRaw` and `requestData`

Reply:
isConnected must not acquire `_lock`, as it seemed to be called under the lock already.